### ISSUE TITLE
Use undefined for uninitialized result&error

### ISSUE THREAD

**Issue by vadzim** [bsonntag/react-use-promise#10]
Please, consider using `undefined` instead of `null` for undefined values of `error` and `result` fields.

Generally, in javascript `undefined` is used for absent or uninitialized values. I understand, that having both `undefined` and `null` in javascript can bother, but this is how things work for now. So `undefined` should be used for consistency.

**Comment by gzaripov** [comment]
One of consequences is that null blocks default values for array destructuring.
 If status is null it wont be assigned to defaultStatus
`const [status = defaultStatus] = usePromise(...)`

so I need to do
```
let [status] = usePromise(...);
status = status || defaultStatus;
```
It is really annoying.

**Comment by bsonntag** [comment]
Released as `0.2.0`.
